### PR TITLE
Restructure EinsplineSet classes

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
1. Relocate files to QMCWaveFunctions/BsplineFactory
2. Remove EinsplineSetBuilderOld.cpp and EinsplineSetBuilderReadBands_ESHDF.cpp as they contain only one function each. Put these function into other EinsplineSet class related files.

## What type(s) of changes does this code introduce?
- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
laptop

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

**Comment by ye-luo** [comment]
Test this please

File: src/QMCWaveFunctions/BsplineFactory/EinsplineSetBuilderESHDF.fft.cpp [hunk 1]
```diff
@@ -339,6 +368,127 @@ bool EinsplineSetBuilder::ReadOrbitalInfo_ESHDF(bool skipChecks)
...
+  if (root)
+  {
+    H5File.readEntry(MeshSize, "/electrons/psi_r_mesh");
+    H5File.readEntry(MeshSize, "/electrons/mesh");
+  }
+  myComm->bcast(MeshSize);
+  hasPsig = (MeshSize[0] == 0);
+  if (hasPsig)
+  {
+    int nallowed  = 257;
```
Comment (prckent) [inline src/QMCWaveFunctions/BsplineFactory/EinsplineSetBuilderESHDF.fft.cpp:386]: Can you explain + justify this please?

**Review by prckent — APPROVED** [review]

File: src/QMCWaveFunctions/BsplineFactory/EinsplineSetBuilderESHDF.fft.cpp (hunk 1 above)
Comment (ye-luo) [reply src/QMCWaveFunctions/BsplineFactory/EinsplineSetBuilderESHDF.fft.cpp:386]: I think this was initially introduced to block bad fft grids in certain libraries (IIRC essl).

Comment (ye-luo) [reply src/QMCWaveFunctions/BsplineFactory/EinsplineSetBuilderESHDF.fft.cpp:386]: This is not new code. I simply moved it from another file which has been dissolved.

**Pull request merged**